### PR TITLE
Feature/read update and delete users data

### PR DESCRIPTION
**Pull request by IMMANUEL5015**
### **Overview**
  This PR allows the application to perform full CRUD operations on user's data, where previously, **CRUD** could only be performed for BucketLists.

### **Notes**
 * An admin can get all users data.
 * An admin can get the data of any specific user.
 * Regular users can get only their own information.
 * An admin can update the data of any specific user.
 * Regular users can update only their own information.
 * An admin can delete the data of any specific user.
 * Regular users can delete only their own information.
 * Users can update their passwords.

### CODE REVIEW

File: server/controllers/authController.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 const { promisify } = require('util');
 const jwt = require('jsonwebtoken');
 const User = require('../models/usermodel');
 const bcrypt = require('bcryptjs');
-const sendEmail = require('../utilities/email');
-const generateResetToken = require('../utilities/generateResetToken');
-const encryptResetToken = require('../utilities/encryptResetToken');
-const tokenHasExpired = require('../utilities/tokenHasExpired');
-const sendErrorMessage = require('../utilities/sendErrorMessage');
+const sendEmail = require('../utilities/accessories/email');
```
Comment (ngozi-ekekwe) [inline server/controllers/authController.js]: Why did you name this `accessories`?

File: server/controllers/authController.js [hunk 2]
```diff
@@ -25,7 +26,7 @@ exports.signup = async(request, response, next) => {
         });
 
         //Login the user immediately
-        return loggingInUsers(newUser._id, 201, response, 'Success', 'User has been Successfully Created');
+        return loginUserAfterSignup(newUser._id, 201, response, 'Success', 'User has been Successfully Created', newUser);
```
Comment (ngozi-ekekwe) [inline server/controllers/authController.js]: can you move the status code to a config file? Looking at this I can't tell if this is a status code or not.

so maybe in your config file you can define an object 

```
status: {
'ok: 200,
'created: 201,
'not_found`: 404` etc..  
}
``` 

Then import it into your code

File: server/controllers/authController.js (hunk 1 above)
Comment (IMMANUEL5015) [reply server/controllers/authController.js]: The files were becoming too much in one folder. I needed to organize them into separate folders. Please, can you suggest a better name?

File: server/controllers/authController.js (hunk 2 above)
Comment (IMMANUEL5015) [reply server/controllers/authController.js]: These are just helper functions. In the underlying code which is currently stored in a separate file, it is very clear that they are status codes.

File: server/controllers/authController.js (hunk 1 above)
Comment (IMMANUEL5015) [reply server/controllers/authController.js]: I'll change the name.

File: server/controllers/authController.js (hunk 2 above)
Comment (IMMANUEL5015) [reply server/controllers/authController.js]: I will make it clearer.

Comment (IMMANUEL5015) [reply server/controllers/authController.js]: Done

File: server/controllers/authController.js (hunk 1 above)
Comment (IMMANUEL5015) [reply server/controllers/authController.js]: The name has been changed.

**Pull request merged**